src/ResourceManager/Network/Commands.Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs

```diff
@@ -71,5 +71,11 @@ public class AzureVirtualNetworkSubnetConfigBase : NetworkBaseCmdlet
             ValueFromPipelineByPropertyName = true,
             HelpMessage = "Service Endpoint Policies")]
         public List<PSServiceEndpointPolicy> ServiceEndpointPolicy { get; set; }
+
+        [Parameter(
```

Comment:
instead of a dlegation ,just add the parameters DelegationName and ServiceName to the New/SetAzureVirtualNetworkSubnetConfigCommand.cs cmdlets.

That way we dont really need a new-**Delegation cmdlet or Add-**Delegation cmdlet.

A user cn add a delegation to a sugnet or set one by passing in theparams.
To make sure that if a user specifies one they specify the pther, add DelegationName and ServiceNAme to a new param set and makethem mandatory there.

Reply:
I don't think this would work well for multiple delegations on a single subnet